Sources/Controllers/Map/Layers/OAMyPositionLayer.mm

```diff
@@ -616,14 +631,29 @@ - (void)refreshMarkersCollection
     }];
 }
 
-- (void) onSettingsChanged
+- (void) refreshCurrentAppModeMarkerCollection
 {
     __weak OAMyPositionLayer *weakSelf = self;
     dispatch_async(dispatch_get_main_queue(), ^{
-        [weakSelf refreshMarkersCollection];
+        OAApplicationMode *currentMode = [OAAppSettings sharedManager].applicationMode.get;
+        NSNumber *order = _markersOrders[currentMode.name];
+        if (order)
+        {
+            [weakSelf generateMarkerCollectionFor:currentMode baseOrder:[order intValue]];
```

Comment:
You must not touch markers in collection until they are used in core!
This is example how all or one marker must be replaced:
```
        [self invalidateMarkersCollection];
        [self generateMarkersCollection];
        [self updateMyLocationCourseProvider];
```

invalidate - generate - update